lib/lockfile.ml

```diff
@@ -364,11 +379,12 @@ let from_opam ~opam_monorepo_cwd ?file opam =
       source_config;
     }
 
-let save ~opam_monorepo_cwd ~file t =
+let save ~opam_monorepo_cwd ~cli_args ~file t =
   let opam = to_opam ~opam_monorepo_cwd t in
+  let with_cli_args = Cli_args.add cli_args opam in
```

Comment:
Wouldn't it be nicer to do something like

```suggestion
  let opam = to_opam ~opam_monorepo_cwd t
    |> Cli_args.add cli_args
  in
```
Then it is clear that `opam` still contains a writeable opam file.